document adjacent docs (sidecar) (#79056)
ansible/ansible: docs/docsite/rst/dev_guide/developing_modules_documenting.rst

  Some older Ansible modules have ``imports`` at the bottom of the file, ``Copyright`` notices with the full GPL prefix, and/or ``DOCUMENTATION`` fields in the wrong order. These are legacy files that need updating - do not copy them into new modules. Over time we are updating and correcting older modules. Please follow the guidelines on this page!

> .. note:: For non Python modules you still create a ``.py`` file for documentation purposes, but starting at ansible-core 2.14 you can opt to create a ``.yml`` file instead that has the same structure of data but in pure YAML.

Reviewer: ```suggestion
.. note:: For non-Python modules you still create a ``.py`` file for documentation purposes. Starting at ansible-core 2.14 you can instead choose to create a ``.yml`` file that has the same data structure, but in pure YAML.
```

>           With YAML files, the examples below are easy to use by removing Python quoting and substituting ``=`` for ``:``, for example ``DOCUMENTATION = r''' ... '''` ` to ``DOCUMENTATION: ...`` and removing closign quotes. :ref:`adjacent_yaml_doc`

Reviewer: ```suggestion
          With YAML files, the examples below are easy to use by removing Python quoting and substituting ``=`` for ``:``, for example ``DOCUMENTATION = r''' ... '''` ` to ``DOCUMENTATION: ...`` and removing closing quotes. :ref:`adjacent_yaml_doc`
```

.. _shebang:

Python shebang & UTF-8 coding
===============================

Begin your Ansible module with ``#!/usr/bin/python`` - this "shebang" allows ``ansible_python_interpreter`` to work. Follow the shebang immediately with ``# -*- coding: utf-8 -*-`` to clarify that the file is UTF-8 encoded.

> .. note:: Using ``#!/usr/bin/env``, makes ``env`` the interpreter and bypasses ``ansilbe_<interpreter>_interpreter`` logic.

Reviewer: ```suggestion
.. note:: Using ``#!/usr/bin/env``, makes ``env`` the interpreter and bypasses ``ansible_<interpreter>_interpreter`` logic.
```

.. note:: If you develop the module using a different scripting language, adjust the interpreter accordingly (``#!/usr/bin/<interpreter>``) so ``ansible_<interpreter>_interpreter`` can work for that specific language.

> .. note:: Binary modules do not require a shebang nor an interpreter.

Reviewer: ```suggestion
.. note:: Binary modules do not require a shebang or an interpreter.
```
